src/sparsezoo/objects/file.py

```diff
@@ -90,6 +92,12 @@ def __init__(
         override_parent_path: Union[str, None] = None,
         **kwargs,
     ):
+        if isinstance(model_metadata, dict):
```

Comment:
should likely have an assert here to fail for the case where release_version is passed in and does not override the model_metadata since that information will be dropped

Reply:
The release version should only override when model_metadata is a dict that does not have release_version. Updated logic to make that more clear.